src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -3935,13 +4018,89 @@ private void DeleteObjectInternal(object entity, bool failIfInAddedState)
                     Util.IncludeLinkState(state),
                     "bad state transition to deleted");
 
+                if (dependsOnIds != null)

```

Comment:
@KenitoInc This block of code mirrors the other in `UpdateObjectInternal`. Refactor into a shared object?

Reply:
As a matter of fact you could redesign and rename `GetDependsOnChangeOrdersAndChangeSetIds` such that you pass `resource` into it and that method populates `DependsOnIds` and `DependsOnChangeSetIds` properties

Reply:
Resolved